### PR TITLE
Fix semicolon edge case involving line comment before `else` keyword

### LINKED ISSUE #143: Formatter adds semicolon after `if` clause in presence of `else`
The formatter seems to automatically append a semicolon after the if clause, even if there is an else clause, as long as they are separated by a newline. See the following example, where a semicolon was inserted after the if clause closing brace:
```
// Case 1: Peak is either here or to the left
if (nums[middle] < nums[middle +1]) {
  right := middle;
};
// Case 2: Peak is somewhere to the right of here
else {
  left := middle + 1;
};
```

### PR DESCRIPTION
**Pull request by rvanasa**
Resolves #143.

### CODE REVIEW

File: src/parsers/motoko-tt-parse/preprocess.ts [hunk 1]
```diff
@@ -78,11 +78,21 @@ export default function preprocess(
                     indent++;
                 }
 
+                // Find next non-empty line index
```
Comment (ggreif) [inline src/parsers/motoko-tt-parse/preprocess.ts:81]: do comment-only lines count as empty?

Comment (ggreif) [reply src/parsers/motoko-tt-parse/preprocess.ts:81]: Oh the test says "yes"!

**Review by ggreif — APPROVED** [review]
Cool, thanks for attacking this!

File: src/parsers/motoko-tt-parse/preprocess.ts (hunk 1 above)
Comment (rvanasa) [reply src/parsers/motoko-tt-parse/preprocess.ts:81]: Oh, yes I'll clarify that. Empty after filtering out comments.

**Pull request merged**